source/backup-cursor.rst

```diff
@@ -0,0 +1,96 @@
+
+They are available in |PSMDB| starting with version 4.4.6-8. 
+
+
+Percona provides `Percona Backup for MongoDB (PBM) <https://www.percona.com/doc/percona-backup-mongodb/index.html>`_ – a light-weight open source solution for consistent backups and restores across sharded clusters. PBM relies on these aggregation stages for physical backups and restores. However, if you wish to develop your own backup application, this document describes the ``$backupCursor`` and ``$backupCursorExtend`` aggregation stages.
+
+Usage 
+=============
+
+You can run these stages in any type of MongoDB deployment. For single node deployment, run the ``$backupCursor``, then the ``$backupCursorExtend`` and save the output files to the backup storage.  
+
+For sharded clusters and non-sharded replica sets, run both aggregation stages on one node from each replica set. It can be either the primary or the secondary node. Note that since the secondary node may lag in syncing the data from the primary one, you will have to wait for the exact same time before running the ``$backupCursorExtend``.
+
+
+Get a list of all files to copy with $backupCursor
+---------------------------------------------------------------
+
+.. code-block:: javascript
+
+    var bkCsr = db.getSiblingDB("admin").aggregate([{$backupCursor: {}}])
+    var bkCsrMetadata = bkCsr.next().metadata
+
+Sample output:
+
+.. code-block:: 
+
+   bkCsrMetadata
+
+   [
+     {
+       metadata: {
+         backupId: UUID("35c34101-0107-44cf-bdec-fad285e07534"),
+         dbpath: '/var/lib/mongodb',
+         oplogStart: { ts: Timestamp({ t: 1666631297, i: 1 }), t: Long("-1") },
+         oplogEnd: { ts: Timestamp({ t: 1666631408, i: 1 }), t: Long("1") },
+         checkpointTimestamp: Timestamp({ t: 1666631348, i: 1 })
+       }
+     },
+     
+
+Store the ``metadata`` document somewhere, because you need to pass the ``backupId`` parameter from this document as the input parameter for the ``$backupCursorExtend`` stage. Also you need the ``oplogEnd`` timestamp.
+Make sure that the ``$backupCursor`` is complete on all shards in your cluster.
+
+.. note::
+
+   Note that when running ``$backupCursor`` in a single node deployment, ``oplogStart``, ``oplogEnd``, ``checkpointTimesatmp`` values may be absent. This is because single node deployments don't have oplogs.   
+
+Run ``$backupCursorExtend`` to retrieve the WiredTiger transaction logs
+--------------------------------------------------------------------------
+
+Pass the ``backupId`` from the metadata document as the first parameter. For the ``timestamp`` parameter, use the maximum (latest) value among the ``oplogEnd`` timestamps from all shards and config server replica set. This will be the target time to restore.
+
+.. code-block:: javascript
+
+   bkExtCsr = db.aggregate([{$backupCursorExtend: {backupId: bkCsrMetadata.backupId, timestamp: new Timestamp(1666631418, 1)}}])
```

Comment:
but we need to return VAR here - my bad, made a comment last minute in a hurry) 

var bkExtCsr = db.aggregate([{$backupCursorExtend: {backupId: bkCsrMetadata.backupId, timestamp: new Timestamp(1666631418, 1)}}])

Reply:
updated